pkg/gadgets/internal/networktracer/tracer.go

```diff
@@ -119,7 +143,13 @@ func NewTracer[Event any](
 	baseEvent func(ev types.Event) Event,
 	parseEvent func([]byte) (*Event, error),
 ) *Tracer[Event] {
+	socketEnricher, err := socketenricher.NewSocketEnricher()
+	if err != nil {
+		// Non fatal: an older kernel might not support bpf extensions
```

Comment:
After commenting on https://github.com/inspektor-gadget/inspektor-gadget/pull/1201/files#r1095999193, I checked this and I think we can fix both issues with the same change: Only create the socker enricher if this is use by the gadget, I tried some experiments in https://github.com/inspektor-gadget/inspektor-gadget/commit/5a74593a841da7ce5b8351e9b73cb67465d3bd2c.

Reply:
The SocketEnricher uses CO-RE, so your patch 5a74593a841da7ce5b8351e9b73cb67465d3bd2c would break the dns and sni gadgets on older kernels without BTF/btfhub support. Is it ok?